src/test/kotlin/com/contentful/java/cma/CMAHttpExceptionTest.kt

```diff
@@ -0,0 +1,232 @@
+package com.contentful.java.cma
+
+import com.contentful.java.cma.lib.TestUtils
+import com.contentful.java.cma.model.CMAHttpException
+import okhttp3.*
+import kotlin.test.assertEquals
+import org.junit.Test as test
+
+class CMAHttpExceptionTest : BaseTest() {
+    val HEADER_RATE_LIMIT_HOUR_LIMIT = "X-Contentful-RateLimit-Hour-Limit"
+    val HEADER_RATE_LIMIT_HOUR_REMAINING = "X-Contentful-RateLimit-Hour-Remaining"
+    val HEADER_RATE_LIMIT_SECOND_LIMIT = "X-Contentful-RateLimit-Second-Limit"
+    val HEADER_RATE_LIMIT_SECOND_REMAINING = "X-Contentful-RateLimit-Second-Remaining"
+    val HEADER_RATE_LIMIT_RESET = "X-Contentful-RateLimit-Reset"
+
+    @test
+    fun testHeadersParsing() {
+        val request = Request.Builder().url("https://example.com/foo").build()
+
+        val response = buildMockedResponse(request, "{\"sys\":{\"id\":\"manualid\"}}")
+
+        val e = CMAHttpException(request, response)
+
+        assertEquals("message", e.responseMessage())
```

Comment:
I'd prefer if the assignment of `"message"` to your `CMAHttpException` instances responseMessage happened in teh scope of this test rather than somewhere in a different file (or many lines away in same file).